### PR TITLE
feat: tally community tool | NPG-5428

### PR DESCRIPTION
**Pull request by cong-or**
Community tool for cross referencing and validating tally results.

### CODE REVIEW

**Comment by cong-or** [comment]
@stevenj using legacy crypto and vote logic as it is cleaner than importing the `vote` and `crypto`crates.

It's the same under the hood regardless and can be easily changed to the newer crates if we want down the line.

**Review by Mr-Leshiy — COMMENTED** [review]
@cong-or but what do you think to extend this tally tool to work also with the `p256k1` backend instead of ristretto ?

**Review by Mr-Leshiy — APPROVED** [review]
Despite few comments LGTM

**Comment by cong-or** [comment]
> @cong-or but what do you think to extend this tally tool to work also with the `p256k1` backend instead of ristretto ?

Ristretto should cover us until we migrate to cat 2.0. I imagine the tool will look a bit different for 2.0 and we can refactor then. I think this should be enough for 1.0

File: src/tally/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    //
+    // Load decrypt shares for cross referencing Tally result
+    // Intended for public use
+    //
+    if let Some(shares) = args.decrypt_tally_from_shares {
+        let shares = load_decrypt_shares(shares).expect("unable to load decrypt shares");
+        let pub_keys = args
+            .public_keys
+            .clone()
+            .expect("key requirement enforced by clap");
```
[the remark this answers is not in the document]
Comment (stevenj) [reply src/tally/src/main.rs:81]: I would rather we not use `expect` as well.  Unless its actually not possible to trigger under reasonable circumstances.
If so, it should be commented such.

**Review by stevenj — DISMISSED** [review]
expects to errors returned, or comment that the expect shouldn't actually trigger in reasonable use (ie user can't give you data which triggers it).

File: src/tally/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cong-or) [reply src/tally/src/main.rs:81]: clap actually enforces the public key requirement so the expect never happens.

**Review by Mr-Leshiy — APPROVED** [review]
LGTM

**Pull request merged**